modules/smartxspBidAdapter.js

```diff
@@ -0,0 +1,120 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+import {find} from '../src/polyfill.js';
+
+const SMARTXSP_CONFIG = {
+  bidRequestUrl: 'https://tag.smartxsp.io/pbjs/bid',
+  method: 'POST'
+}
+
+const BIDDER_CODE = 'smartxsp';
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+  isBidRequestValid: function (bid) {
+    return !!(bid && bid.params && bid.params.accountId && bid.params.widgetId);
+  },
+  buildRequests: function (validBidRequests, bidderRequest) {
+    var i
+    var j
+    var bid
+    var bidParam
+    var bidParams = []
+    var sizes
+    var frameWidth = Math.round(window.screen.width)
+    var frameHeight = Math.round(window.screen.height)
+    for (i = 0; i < validBidRequests.length; i++) {
+      bid = validBidRequests[i]
+      if (bid.sizes) {
+        sizes = bid.sizes
+      } else if (typeof (BANNER) != 'undefined' && bid.mediaTypes && bid.mediaTypes[BANNER] && bid.mediaTypes[BANNER].sizes) {
+        sizes = bid.mediaTypes[BANNER].sizes
+      } else if (frameWidth && frameHeight) {
+        sizes = [[frameWidth, frameHeight]]
+      } else {
+        sizes = []
+      }
+      for (j = 0; j < sizes.length; j++) {
+        bidParam = {
+          accountId: bid.params.accountId || '',
+          bidId: bid.bidId,
+          'banner-format-width': sizes[j][0],
+          'banner-format-height': sizes[j][1]
+        }
+        if (bid.params.bannerFormat) {
+          bidParam['banner-format'] = bid.params.bannerFormat
+        }
+        if (bid.params.language) {
+          bidParam.language = bid.params.language
```

Comment:
read from standard locations

Reply:
this is not resolved, you should read language from the openrtb location eg how yahoo does here https://github.com/prebid/Prebid.js/blob/509deaa0649824369907e6b76d0a4d045ec84df8/modules/yahoosspBidAdapter.js#L442